src/app/chat/[uuid]/page.tsx

```diff
@@ -1,23 +1,28 @@
 "use client";
 
 import { notFound, useParams } from "next/navigation";
+import { useEffect, useState } from "react";
 
+import { useChats } from "@/hooks/use-chats";
 import type { Chat } from "@/types/chat";
 
 export default function ConversationPage() {
   const parameters = useParams<{ uuid: string }>();
+  const { getChat } = useChats();
+  const [chat, setChat] = useState<Chat | null | undefined>(null);
 
-  // Get chats from local storage
-  const chats =
-    localStorage.getItem("chats") == null
-      ? []
-      : (JSON.parse(localStorage.getItem("chats")!) as Chat[]);
+  useEffect(() => {
+    const foundChat = getChat(parameters.uuid);
+    setChat(foundChat);
 
-  // Find chat by uuid
-  const chat = chats.find((c) => c.uuid === parameters.uuid);
+    //TODO call api there
+  }, [parameters.uuid, getChat]);
 
-  // If chat not found, return 404
-  if (!chat) {
+  if (chat === null) {
+    return;
+  }
+
+  if (chat === undefined) {
     notFound();
   }
```

Comment:
**issue:** This whole section needs to be refactored.

Code within `if (chat === undefined)` will *never* be called. Local storage response is either the saved value or null, if no value tied to the given key is found. There's no way for this condition to be met as of now. You can check it by going to `localhost:3000/chat/absolutely-anything-here` to see that no 404 page is shown as the `if (chat === null)` block is called, and that one just returns - resulting in us seeing just a blank page

Our chat page can be in three states
1. Awaiting model response (chat found && no model response already)
2. Showing the suggestions (chat found && model response exists)
3. Showing the 404 page (chat not found)

My suggestion is to remove the two marked `if` blocks and approach this differently: 
Redirect to 404 in the useEffect itself. If `targetChat` is `null` then that means a chat with that uuid *was not* found within the local storage
```tsx
useEffect(() => {
  const targetChat = getChat(parameters.uuid);
  if (targetChat === null) { notFound() }
  setChat(targetChat);
  // ...rest of the code
```
since `null` is the initial value of `chat` state, use it to determine whether we're loading the chat 
```tsx
return <div>{chat ? chat.prompt : "Loading"}</div>;
```